akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -100,15 +106,43 @@ private[stream] object Timers {
     override def toString = "IdleTimeout"
   }
 
+  final class BackpressureTimeout[T](timeout: FiniteDuration) extends SimpleLinearGraphStage[T] {
+    override def initialAttributes = DefaultAttributes.backpressureTimeout
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) {
+      private var nextDeadline: Option[Deadline] = None
+
+      setHandler(in, new InHandler {
+        override def onPush(): Unit = {
+          push(out, grab(in))
+          nextDeadline = Some(Deadline.now + timeout)
```

Comment:
I think we should optimize this to avoid all those allocations, i.e. using nanoseconds in a `Long` field